common/io/io.cpp

```diff
@@ -92,50 +92,93 @@ Error Copy(Writer &dst, Reader &src, vector<uint8_t> &buffer) {
 	}
 }
 
-void AsyncCopy(Writer &dst, AsyncReader &src, function<void(Error)> finished_handler) {
+struct CopyData {
+	CopyData(size_t limit) :
+		buf(MENDER_BUFSIZE),
+		limit(limit) {
+	}
+
+	vector<uint8_t> buf;
+	size_t copied {0};
+	size_t limit;
+};
+
+void AsyncCopy(
+	Writer &dst, AsyncReader &src, function<void(Error)> finished_handler, size_t stop_after) {
 	AsyncCopy(
 		WriterPtr(&dst, [](Writer *) {}),
 		AsyncReaderPtr(&src, [](AsyncReader *) {}),
 		finished_handler);
```

Comment:
Missing to pass `stop_after` to `AsyncCopy`.

Reply:
Embarrassed! :see_no_evil: Thanks!